### PR TITLE
Add config printer to janusgraph-server.sh

### LINKED ISSUE #2857: The jvm.options feature of janusgraph-server.sh is broken
Since janusgraph-0.6.0 the starting script, renamed to `bin/janusgraph-server.sh`, reads the `JAVA_OPTIONS` from the file `conf/jvm.options`. As [indicated by user m.leinweber@..,](https://lists.lfaidata.foundation/g/janusgraph-users/topic/jvm_options_broken/86896083) the reading of these `jvm.options` file is broken. Instead of reading all options, only the very last option is applied.

# Steps to reproduce
Copy `bin/janusgraph-server.sh` to `bin/janusgraph-server-tmp.sh` and add the following debug lines starting from line 110:
```
echo $JAVA_OPTIONS_FILE
echo $JAVA_OPTIONS
exit 1
```

# Workaround
Edit the `bin/janusgraph-server.sh` file manually and replace line 103 reading:  
```
  for opt in `grep "^-" $JAVA_OPTIONS_FILE`
```
with  
```
  for opt in "`grep '^-' $JAVA_OPTIONS_FILE`"
```
As often in failing bash scripts, quoting was done in an improper way.

### PR DESCRIPTION
**Pull request by farodin91**
Fixes #2857

Signed-off-by: Jan Jansen <jan.jansen@gdata.de>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### CODE REVIEW

File: janusgraph-dist/src/assembly/static/bin/janusgraph-server.sh [hunk 1]
```diff
@@ -254,8 +260,18 @@ startForeground() {
   fi
 }
 
+printConfig() {
+    echo "JANUSGRAPH_HOME: ${JANUSGRAPH_HOME}"
+    echo "JANUSGRAPH_YAML: ${JANUSGRAPH_YAML}"
+    echo "JANUSGRAPH_SERVER_CMD: ${JANUSGRAPH_SERVER_CMD}"
+    echo "LOG_DIR: ${LOG_DIR}"
+    echo "PID_FILE: ${PID_FILE}"
+    echo "JAVA_OPTIONS: ${JAVA_OPTIONS}"
```
Comment (FlorianHockmann) [inline janusgraph-dist/src/assembly/static/bin/janusgraph-server.sh:269]: This should print the full `JAVA_OPTIONS` including everything from `jvm-11.options` / ``jvm-8.options`, right? I just build the distribution archive on my machine from this branch and then executed the script to get the output produced here, but the `JAVA_OPTIONS` don't look very complete:

```bash
$ bin/janusgraph-server.sh config
JANUSGRAPH_HOME: /mnt/c/Workspace/GitHub/janusgraph/janusgraph-dist/target/janusgraph-full-0.6.1-SNAPSHOT
JANUSGRAPH_YAML: /mnt/c/Workspace/GitHub/janusgraph/janusgraph-dist/target/janusgraph-full-0.6.1-SNAPSHOT/conf/gremlin-server/gremlin-server.yaml
JANUSGRAPH_SERVER_CMD: org.janusgraph.graphdb.server.JanusGraphServer
LOG_DIR: /mnt/c/Workspace/GitHub/janusgraph/janusgraph-dist/target/janusgraph-full-0.6.1-SNAPSHOT/logs
PID_FILE: /mnt/c/Workspace/GitHub/janusgraph/janusgraph-dist/target/janusgraph-full-0.6.1-SNAPSHOT/run/janusgraph.pid
JAVA_OPTIONS:  -XX:+CMSClassUnloadingEnabled  -javaagent:/mnt/c/Workspace/GitHub/janusgraph/janusgraph-dist/target/janusgraph-full-0.6.1-SNAPSHOT/lib/jamm-0.3.0.jar
COLLECTED_JAVA_OPTIONS_FILE:  -XX:+CMSClassUnloadingEnabled
```

This is still just the last line from the `OPTIONS` file `XX:+CMSClassUnloadingEnabled` together with the `javaagent`, isn't it?

File: janusgraph-dist/src/assembly/static/bin/janusgraph-server.sh [hunk 2]
```diff
@@ -254,8 +260,18 @@ startForeground() {
...
+    echo "COLLECTED_JAVA_OPTIONS_FILE: ${COLLECTED_JAVA_OPTIONS_FILE}"
```
Comment (FlorianHockmann) [inline janusgraph-dist/src/assembly/static/bin/janusgraph-server.sh:270]: Here, I would expect to get a file path as this ends on `FILE` but I'm actually getting its content:

```
COLLECTED_JAVA_OPTIONS_FILE:  -XX:+CMSClassUnloadingEnabled
```

is that supposed to be like that? If so, then I think that the name is confusing. I'm also not sure if we really want to include this output here. Isn't that only useful to debug the problem this is supposed to fix?

File: janusgraph-dist/src/test/java/org/janusgraph/pkgtest/JanusGraphAssemblyBaseIT.java [hunk 3]
```diff
@@ -128,6 +128,9 @@ protected static void unzip(String dir, String zipFile) throws IOException, Inte
 
     protected static void command(File dir, String... command) throws IOException, InterruptedException {
         ProcessBuilder pb = new ProcessBuilder(command);
+
+        Map<String, String> env = pb.environment();
```
Comment (FlorianHockmann) [inline janusgraph-dist/src/test/java/org/janusgraph/pkgtest/JanusGraphAssemblyBaseIT.java:132]: Why was this change necessary? Is that related to the changes in `janusgraph-server.sh`?

File: janusgraph-dist/src/assembly/static/conf/jvm-11.options [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+#  GC SETTINGS  #
+#################
+
+### CMS Settings
+
+-XX:+UseG1GC
+-XX:+CMSParallelRemarkEnabled
+-XX:SurvivorRatio=8
+-XX:MaxTenuringThreshold=1
+-XX:CMSInitiatingOccupancyFraction=75
```
Comment (FlorianHockmann) [inline janusgraph-dist/src/assembly/static/conf/jvm-11.options:72]: Aren't all these `CMS[...]` values config options of the `ConcMarkSweep` garbage collector that we had configured for Java 8, but not that we use G1 instead they shouldn't have any effect, right?

**Review by FlorianHockmann — CHANGES_REQUESTED** [review]

File: janusgraph-dist/src/test/java/org/janusgraph/pkgtest/JanusGraphAssemblyBaseIT.java (hunk 3 above)
Comment (farodin91) [reply janusgraph-dist/src/test/java/org/janusgraph/pkgtest/JanusGraphAssemblyBaseIT.java:132]: Yeah, it now reads the value from options. It would set the memory to 4G and github actions doesn't have enough memory/

File: janusgraph-dist/src/assembly/static/bin/janusgraph-server.sh (hunk 1 above)
Comment (farodin91) [reply janusgraph-dist/src/assembly/static/bin/janusgraph-server.sh:269]: I will check again. Could be my last commit to fix codacy issues.

File: janusgraph-dist/src/assembly/static/bin/janusgraph-server.sh (hunk 2 above)
Comment (farodin91) [reply janusgraph-dist/src/assembly/static/bin/janusgraph-server.sh:270]: Will remove this log as they are already include in JAVA_OPTIONS

File: janusgraph-dist/src/assembly/static/conf/jvm-11.options (hunk 4 above)
Comment (farodin91) [reply janusgraph-dist/src/assembly/static/conf/jvm-11.options]: lines are removed

**Comment by farodin91** [comment]
@FlorianHockmann Would you to review it again?

**Review by FlorianHockmann — APPROVED** [review]
Just checked again and it all looks good now.

**Comment by farodin91** [comment]
@FlorianHockmann I will change the base branch to 0.6.
